x/cdp/keeper/draw.go

```diff
@@ -175,6 +175,13 @@ func (k Keeper) ValidatePaymentCoins(ctx sdk.Context, cdp types.CDP, payment sdk
 		}
 		return types.ErrInvalidPaymentDenom(k.codespace, cdp.ID, principalDenoms, paymentDenoms)
 	}
+	for _, dc := range payment {
+		dp, _ := k.GetDebt(ctx, dc.Denom)
+		proposedBalance := cdp.Principal.AmountOf(dc.Denom).Sub(dc.Amount)
+		if proposedBalance.GT(sdk.ZeroInt()) && proposedBalance.LT(dp.DebtFloor) {
```

Comment:
doesn't catch case where payment is > principal
looks like it's not caught elsewehere, so someone could over repay